client/chain/chain.go

```diff
@@ -917,9 +979,19 @@ func (c *chainClient) SpotOrder(defaultSubaccountID eth.Hash, network common.Net
 	}
 }
 
+// Deprecated: Use CreateDerivativeOrder instead
 func (c *chainClient) DerivativeOrder(defaultSubaccountID eth.Hash, network common.Network, d *DerivativeOrderData) *exchangetypes.DerivativeOrder {
 
-	market, isPresent := c.marketsAssistant.AllDerivativeMarkets()[d.MarketId]
+	assistant, err := NewMarketsAssistant(network.Name)
+	if err != nil {
+		panic(err)
+	}
+
+	return c.CreateDerivativeOrder(defaultSubaccountID, network, d, assistant)
+}
+
+func (c *chainClient) CreateDerivativeOrder(defaultSubaccountID eth.Hash, network common.Network, d *DerivativeOrderData, marketAssistant MarketsAssistant) *exchangetypes.DerivativeOrder {
+	market, isPresent := marketAssistant.AllDerivativeMarkets()[d.MarketId]
```

Comment:
In AllDerivativeMarkets(), we don't need to clone the map()

Reply:
Removed the clone code